plugin-server/src/main/ingestion-queues/session-recording/session-recordings-blob-consumer.ts

```diff
@@ -132,6 +133,9 @@ export class SessionRecordingBlobIngester {
             return statusWarn('message value or timestamp is empty')
         }
 
+        // track the latest message timestamp seen so we can use it to calculate a reference "now"
+        this.latestKafkaMessageTimestamp = message.timestamp
```

Comment:
Do we know for sure that the timestamps are in order 🤔  wondering if we potentially receive the kafka messages unordered for some reason which would also explain why the offsets were potentially out of order...

Reply:
I wondered about that... 

My assumption was that even if they're out of order they're close enough in time that tracking last received is a good proxy for "kafka now" but 🙈